metrics/collectors/images/images_collector.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2021 Canonical Ltd
+
+from metrics.lib.basemetric import Metric
+from launchpadlib.launchpad import Launchpad
+
+import requests, datetime, tempfile
+
+DESKTOP_CURRENT_DAILY = "http://cdimage.ubuntu.com/daily-live/current/%s-desktop-%s.iso"
+DESKTOP_LTS_DAILY = "http://cdimage.ubuntu.com/%s/daily-live/current/%s-desktop-%s.iso"
+
+class ImagesMetrics(Metric):
+    def __init__(self, dry_run=False, verbose=False):
+        super().__init__(dry_run, verbose)
+
+        self.lp = Launchpad.login_anonymously(
+            "metrics",
+            "production",
+            launchpadlib_dir=tempfile.mkdtemp(),
```

Comment:
just a note, `UbuntuQueueMetrics` has the same problem and this can be fixed later. This temp dir might not be cleaned up. On the actual server we run with `PrivateTmp=yes` but if you're testing locally then directories are leaked.

One thought is to run the metrics under `with closing` and let them define a `close()` function which can do any cleanups, or do the same in `run()`, and call `close()` at the end.

Reply:
https://github.com/ubuntu/ubuntu-release-metrics/issues/8